tests/dtypes_test.py

```diff
@@ -133,12 +133,22 @@ def test_struct_hashes() -> None:
     assert len({hash(tp) for tp in (dtypes)}) == 3
 
 
-@pytest.mark.skipif(PANDAS_VERSION < (2, 2), reason="old pandas")
-def test_2d_array(constructor: Constructor, request: pytest.FixtureRequest) -> None:
-    if any(x in str(constructor) for x in ("dask", "modin", "cudf", "pyspark")):
-        request.applymarker(pytest.mark.xfail)
-    if "pyarrow_table" in str(constructor) and PYARROW_VERSION < (14,):
-        request.applymarker(pytest.mark.xfail)
+def test_2d_array(constructor: Constructor) -> None:
+    version_conditions = [
+        (PANDAS_VERSION < (2, 2), "Requires pandas 2.2+ for 2D array support"),
+        (
+            any(x in str(constructor) for x in ("dask", "modin", "cudf", "pyspark")),
+            "2D array operations not supported in these backends",
```

Comment:
in theory i think these should be able to support 2D arrays, shall we keep these to xfail?